mlflow/genai/utils/trace_utils.py

```diff
@@ -243,10 +247,37 @@ def extract_expectations_from_trace(
     return {exp.name: exp.expectation.value for exp in expectation_assessments}
 
 
+def _wrap_async_predict_fn(async_fn: Callable[..., Any]) -> Callable[..., Any]:
+    """
+    Wrap an async function to make it synchronous using asyncio.run with timeout.
+
+    Args:
+        async_fn: The async function to wrap
+
+    Returns:
+        A synchronous wrapper function that calls the async function with timeout
+    """
+    timeout = MLFLOW_GENAI_EVAL_ASYNC_TIMEOUT.get()
+
+    @functools.wraps(async_fn)
+    def sync_wrapper(*args, **kwargs):
+        return asyncio.run(asyncio.wait_for(async_fn(*args, **kwargs), timeout=timeout))
```

Comment:
Has this been tested in a Jupyter / Databricks notebook context (just a cautionary confirmation here to ensure that the global event loop handles this properly) ?

Reply:
Good call! Added support for this and tested it [here](https://e2-dogfood.staging.cloud.databricks.com/editor/notebooks/4054488334270244?o=6051921418418893).